test/Log/LogServiceTest.php

```diff
@@ -512,13 +539,20 @@ protected function removeFolder($path)
         }
     }
 
+    protected function deleteFolder($pathName)
```

Comment:
What is the purpose of this function? It only works for empty directories.

Reply:
There were a few errors showing up in the test for "undefined function `deleteFolder()`". From `removeFolder()`, I see you are iterating in the directory files, and passing them into this function. Hence, this function, to delete the files. Atleast, that's the idea if I haven't got the code wrong.

Reply:
Okay, so the actual bug is that in line 532 the name of the function is wrong. Since we need to delete folders recursively in case they contain files or folders, we need to do all that stuff in `removeFolder` and we would still have a problem with `deleteFolder` because we don't know if that folder will be emtpy. So yes, there is bug, but adding `deleteFolder` isn't the solution.

Reply:
Then I think I should first check the files in the directory before calling `rmdir()`? I think adding `$files = new RecursiveIteratorIterator($iterator, RecursiveIteratorIterator::CHILD_FIRST);` after line 533 and calling `rmdir($file->getRealpath()`  for each files will delete the folder along with files in it.